### PR TITLE
Add rules for plugin states to enable plugin saving in AP [#174286329]

### PR DESCRIPTION
**Pull request by dougmartin**
NOTE: The existing answer rule was updated to match what was live on report-service-dev.

### CODE REVIEW

**Review by knowuh — APPROVED** [review]
Looks good to me

File: firestore.rules [hunk 1]
```diff
@@ -17,27 +17,40 @@ service cloud.firestore {
       allow read: if request.auth.token.user_type == 'teacher' && request.auth.token.class_hash == resource.data.context_id;
       allow read: if request.auth.token.user_type == 'learner' && string(request.auth.token.platform_user_id) == string(resource.data.platform_user_id);
       allow update: if request.auth.token.user_type == 'learner'
-              && string(request.auth.token.platform_user_id) == string(resource.data.platform_user_id)
+      				&& string(request.auth.token.platform_user_id) == string(resource.data.platform_user_id)
```
Comment (scytacki) [inline firestore.rules:20]: Looks like you are using tabs here can you switch it back to spaces?

File: firestore.rules [hunk 2]
```diff
@@ -17,27 +17,40 @@ service cloud.firestore {
...
               && request.auth.token.class_hash == resource.data.context_id;
       allow create: if request.auth.token.user_type == 'learner'
-              && string(request.auth.token.platform_user_id) == string(request.resource.data.platform_user_id)
+      				&& string(request.auth.token.platform_user_id) == string(request.resource.data.platform_user_id)
               && request.auth.token.class_hash == request.resource.data.context_id;
       // anonymous requests based on run_key, which must be larger than 10 chars
-      allow read, update: if resource.data.run_key.size() > 10 && resource.data.run_key == request.resource.data.run_key;
-      allow create: if request.resource.data.run_key.size() > 10;
-    }  
+      allow read, update: if resource.data.run_key.size() > 10
```
Comment (scytacki) [inline firestore.rules:26]: I think we disabled this check for testing while developing the anonymous report, and then we forgot to find a better way.  I added a note to this story about fixing this: https://www.pivotaltracker.com/story/show/174883954

**Review by scytacki — APPROVED** [review]
Seems fine to me.
I did find a mix of tabs and spaces, but that isn't crucial to fix.

Also I will want to hold up the release of this to production until this story is complete:
https://www.pivotaltracker.com/story/show/174883954

I started an audit of these rules a little while ago and never finished it. I found several holes.

**Comment by scytacki** [comment]
@dougmartin feel free to merge whenever. Perhaps at the same time you update the report-service-dev rules?

**Pull request merged**